CIP-????/README.md

```diff
@@ -0,0 +1,250 @@
+
+```
+A1185EA2007450696E656170706C6573206F6E2070697A7A613F018263796573
+626E6F
+```
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  , ? 4: vrf_witness
+  , ? 5: ed25519_witness
+  }
+
+question_hash =
+  bytes .size 32
+
+vrf_witness =
+  [ $vrf_vkey, vrf_proof ]
+
+vrf_proof =
+  [ bytes .size 64, bytes .size 16 ]
+
+ed25519_witness =
+  [ $vkey, $signature ]
+```
+
+Some remarks:
+
+1. The field `2` (`question_hash`) is a blake2b-256 hash digest, whose preimage is the entire serialised question metadata payload (with the `94` top-level label).
+1. The fields `4` and `5` are not mutually optional; said differently, at least one of the two must be provided.
+1. The `$vrf_vkey`, `$vkey` and `$signature` are as defined in the original [Cardano ledger specifications](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/real/crypto.cddl) with the usual rules applying to transaction metadata (i.e. byte string longer than 64 bytes are encoded as lists)
+1. In either case (VRF or Ed25519), the witness' signature is calculated from the blake2b-256 hash digest of the serialised `answer` restricted to fields `2` and `3`.
+
+For example, in [CBOR-diagnostic](https://www.rfc-editor.org/rfc/rfc8949#name-diagnostic-notation) notation:
+
+```cbor
+{
+  94: {
+    2: h'3d4854aed3e31d0f5b6637fcd73e8713d9eee86a81d20598f6cb88d9b2bc0a39',
+    3: 2,
+    4:
+      [ h'2dc2fa217af8b52251c4cdf538fa106cbf0b5beac3e74d05f97ceb33c0147a2c'
+      , [ h'056fa6393edbc2f32efe08b9cc455e67bdef6831a709794133cd28fff99b6693069fd27985f023bc419f303d8db998ec774c89e6763518c303f5afbaac31c9e0'
+        , h'2b5358a74c540d6e0e3dac47eff4c604'
+        ]
+      ]
+  }
+}
+```
+
+Which serialises down to:
+
+```
+A1185EA30258203D4854AED3E31D0F5B6637FCD73E8713D9EEE86A81D20598F6
+CB88D9B2BC0A390302048258202DC2FA217AF8B52251C4CDF538FA106CBF0B5B
+EAC3E74D05F97CEB33C0147A2C825840056FA6393EDBC2F32EFE08B9CC455E67
+BDEF6831A709794133CD28FFF99B6693069FD27985F023BC419F303D8DB998EC
+774C89E6763518C303F5AFBAAC31C9E0502B5358A74C540D6E0E3DAC47EFF4C6
+04
+```
+
+> **Warning**
+>
+> Only the first answer to a poll for each credential shall be considered. If multiple answers are found, only the first answer submitted (transaction & block ordering tallying) shall be considered **unless** previous answers are from VRF keys and the latter from an associated cold key. In other words, it is possible to override a VRF-authenticated reply with a cold-key-authenticated one.
+
+### Duration
+
+A poll starts when a valid transaction with a question is posted on-chain and ends at the end of the following epoch, so there's always at least one entire epoch for answering the poll.
+
+In particular, the stake distribution of pools will be considered at the moment the poll is _started_.
```

Comment:
Since a question can be arbitrarily started at any point in the chain, should the most recent snapshot value be used for tabulating stake distribution? Or would this require auditors to perform a "live" stake snapshot at the precise block height of the question hitting the chain?

Reply:
That's a good question; and I guess what I meant to say here is that we consider the stake distribution for the _epoch_ where the poll is started. Stake is always only active after 2 epochs; so there's no point really looking at any "live" stake. Rather, stick to the same 'stake distribution' as considered by the ledger for various calculation.